tests/hazmat/primitives/test_dh.py

```diff
@@ -74,31 +84,268 @@ def test_dh_numbers():
 
 
 def test_dh_parameter_numbers_equality():
-    assert dh.DHParameterNumbers(65537, 3) == dh.DHParameterNumbers(65537, 3)
-    assert dh.DHParameterNumbers(6, 3) != dh.DHParameterNumbers(65537, 3)
-    assert dh.DHParameterNumbers(65537, 0) != dh.DHParameterNumbers(65537, 3)
-    assert dh.DHParameterNumbers(65537, 0) != object()
+    assert dh.DHParameterNumbers(65537, 2) == dh.DHParameterNumbers(65537, 2)
+    assert dh.DHParameterNumbers(6, 2) != dh.DHParameterNumbers(65537, 2)
+    assert dh.DHParameterNumbers(65537, 5) != dh.DHParameterNumbers(65537, 2)
+    assert dh.DHParameterNumbers(65537, 2) != object()
 
 
 def test_dh_private_numbers_equality():
-    params = dh.DHParameterNumbers(65537, 3)
+    params = dh.DHParameterNumbers(65537, 2)
     public = dh.DHPublicNumbers(1, params)
     private = dh.DHPrivateNumbers(2, public)
 
     assert private == dh.DHPrivateNumbers(2, public)
     assert private != dh.DHPrivateNumbers(0, public)
     assert private != dh.DHPrivateNumbers(2, dh.DHPublicNumbers(0, params))
     assert private != dh.DHPrivateNumbers(
-        2, dh.DHPublicNumbers(1, dh.DHParameterNumbers(65537, 0))
+        2, dh.DHPublicNumbers(1, dh.DHParameterNumbers(65537, 5))
     )
     assert private != object()
 
 
 def test_dh_public_numbers_equality():
-    params = dh.DHParameterNumbers(65537, 3)
+    params = dh.DHParameterNumbers(65537, 2)
     public = dh.DHPublicNumbers(1, params)
 
     assert public == dh.DHPublicNumbers(1, params)
     assert public != dh.DHPublicNumbers(0, params)
-    assert public != dh.DHPublicNumbers(1, dh.DHParameterNumbers(65537, 0))
+    assert public != dh.DHPublicNumbers(1, dh.DHParameterNumbers(65537, 5))
     assert public != object()
+
+
+@pytest.mark.requires_backend_interface(interface=DHBackend)
+class TestDH(object):
+    def test_small_key_generate_dh(self, backend):
+        with pytest.raises(ValueError):
+            dh.generate_parameters(2, 511, backend)
+
+    def test_unsupported_generator_generate_dh(self, backend):
+        with pytest.raises(ValueError):
+            dh.generate_parameters(7, 512, backend)
+
+    def test_dh_parameters_supported(self, backend):
+        assert backend.dh_parameters_supported(23, 5)
+        assert not backend.dh_parameters_supported(23, 18)
+
+    def test_convert_to_serialized(self, backend):
```

Comment:
Could we change the tests with the name "serialized" to say "numbers" instead? In general we tend to use serialization for PEM/DER serialization to bytes rather than the *Numbers integer repr classes